### PR TITLE
[BuildPlan] Moved -Xcxx flags to the end of the chain of arguments

### PR DESCRIPTION
**Pull request by nettlep**
- https://bugs.swift.org/browse/SR-6164

### CODE REVIEW

File: Sources/Build/BuildPlan.swift [hunk 1]
```diff
@@ -197,6 +192,11 @@ public final class ClangTargetDescription {
         args += additionalFlags
         args += moduleCacheArgs
         args += buildParameters.flags.cCompilerFlags
+
+        // Add extra C++ flags if this target contains C++ files.
+        if clangTarget.isCXX {
```
Comment (CodaFi) [inline Sources/Build/BuildPlan.swift]: Please document the ordering constraint too.

Comment (nettlep) [reply Sources/Build/BuildPlan.swift]: I've added comments for the change in this PR, but also added comments for the previous PR (https://github.com/apple/swift-package-manager/pull/1365) which covered -Xswiftc, -Xlinker and -Xcc to highlight the fact that those options should go last to enable user overrides of the generated args.

**Comment by aciidb0mb3r** [comment]
Thank you!

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Pull request merged**